e2e/interchaintestv8/relayer_test.go

```diff
@@ -1190,6 +1192,147 @@ func (s *RelayerTestSuite) ICS20TransferERC20TokenBatchedFilteredAckToCosmosTest
 	}))
 }
 
+func (s *RelayerTestSuite) TestUpdateClientToCosmos() {
+	if os.Getenv(testvalues.EnvKeyEthTestnetType) != testvalues.EthTestnetTypePoS {
+		s.T().Skip("Test is only relevant for PoS networks")
+	}
+
+	ctx := context.Background()
+
+	s.SetupSuite(ctx, operator.ProofTypeGroth16) // Doesn't matter, since we won't relay to eth in this test
+
+	eth, simd := s.EthChain, s.CosmosChains[0]
+
+	var initialHeight uint64
+	s.Require().True(s.Run("Get the initial height", func() {
+		resp, err := e2esuite.GRPCQuery[clienttypes.QueryClientStateResponse](ctx, simd, &clienttypes.QueryClientStateRequest{
+			ClientId: testvalues.FirstWasmClientID,
+		})
+		s.Require().NoError(err)
+		s.Require().NotNil(resp.ClientState)
+
+		var wasmClientState ibcwasmtypes.ClientState
+		err = proto.Unmarshal(resp.ClientState.Value, &wasmClientState)
+		s.Require().NoError(err)
+		s.Require().NotZero(wasmClientState.LatestHeight.RevisionHeight)
+
+		initialHeight = wasmClientState.LatestHeight.RevisionHeight
+	}))
+
+	s.Require().True(s.Run("Wait for finality", func() {
+		err := testutil.WaitForCondition(30*time.Minute, 5*time.Second, func() (bool, error) {
+			resp, err := eth.BeaconAPIClient.GetFinalityUpdate()
+			if err != nil {
+				return false, err
+			}
+
+			// resp.Data.Message.Slot is a string, so we need to convert it to a uint64
+			finalizedSlot, err := strconv.ParseUint(resp.Data.FinalizedHeader.Beacon.Slot, 10, 64)
+			if err != nil {
+				return false, err
+			}
+
+			return finalizedSlot > initialHeight, nil
+		})
+		s.Require().NoError(err)
+	}))
+
+	time.Sleep(2 * time.Minute)
```

Comment:
Like we discussed, since this is needed for the client to exist in the finality update, we should probably have a better way to determine this. This test errored at least once with a 2 minute sleep here locally, so this might not be reliable.

Reply:
done